tst/EnergyPlus/unit/RootFinder.unit.cc

```diff
@@ -0,0 +1,176 @@
+    xValue = xMax;
+    yValue = -100.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // Set 3rd point
+    xValue = 20.0;
+    yValue = -1.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // 4th point should converge
+    xValue = rootFinderData.XCandidate;
+    yValue = absoluteYTolerance / 2.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusOK);
+    EXPECT_TRUE(isDone);
+}
+
+TEST_F(EnergyPlusFixture, RootFinder_CheckBracketRoundOff)
+{
+    DataRootFinder::RootFinderDataType rootFinderData;
+    int slopeType = DataRootFinder::iSlopeDecreasing;
+    int methodType = DataRootFinder::iMethodBrent;
+    Real64 relativeXTolerance = 0.0;
+    Real64 absoluteXTolerance = 1.0e-6;
+    Real64 absoluteYTolerance = 1.0e-5;
+
+    RootFinder::SetupRootFinder(rootFinderData, slopeType, methodType, relativeXTolerance, absoluteXTolerance, absoluteYTolerance);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+
+    Real64 xMin = 0.0;
+    Real64 xMax = 100.0;
+    RootFinder::InitializeRootFinder(rootFinderData, xMin, xMax);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+
+    bool isDone = false;
+
+    // Set min point
+    Real64 xValue = xMin;
+    Real64 yValue = 100.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // Set max point
+    xValue = xMax;
+    yValue = -100.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // Set 3rd point (this will be the upper point)
+    xValue = 20.0;
+    yValue = -10.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // Set 4th point slightly below 3rd point but with opposite sign for y (this will be the lower point)
+    xValue = xValue - absoluteXTolerance / 3.0;
+    yValue = 10.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusNone);
+    EXPECT_FALSE(isDone);
+
+    // Set 5th point in between upper and lower, but non-convergent so it gets to the Increment check)
+    xValue = rootFinderData.XCandidate;
+    yValue = 5.0;
+    RootFinder::IterateRootFinder(rootFinderData, xValue, yValue, isDone);
+    EXPECT_EQ(rootFinderData.StatusFlag, DataRootFinder::iStatusOKRoundOff);
+    EXPECT_TRUE(isDone);
+}
```

Comment:
It's really great to see unit tests that directly test the solver tools like this.  So important, and amazing that this is the first set of tests added for RootFinder.cc.  :+1:

Reply:
Yes, and this could be expanded, but this file has fairly good coverage already (other than error paths) as a side-effect of other unit tests. There's also a couple of parallel universes for this with multiple similar methods in [TempSolveRoot](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/TempSolveRoot.cc) and [General::SolveRoot](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/General.cc#L124). I found these during a side-track attempt to do away with Controller:WaterCoil and make `RootFinder` obsolete. Some day . . .